File: JAVA codes/ArmstrongNumberRange.java
//Program to print Armstrong Numbers in a given range
import java.util.Scanner;

public class ArmstrongNumberRange
{
  public static void main(String[] args)
  {
    Scanner in = new Scanner(System.in);
    System.out.println("Armstrong Numbers in a Range");
    System.out.println("^^^^^^^^^^^^^^^^^^^^^^^^");
    System.out.print("Enter the lower limit : ");
    int a = in.nextInt();
    System.out.print("Enter the upper limit : ");
    int b = in.nextInt();
    boolean flag = true;
    for(int i = a; i<= b; i++)
    {
      int sum = 0, length = 0, temp = i;
      //finding the number of digits
      for(length=0; temp != 0; length++, temp /= 10);
      temp = i;
      while(temp!=0)
      {
        sum += Math.pow(temp%10,length);
        temp /= 10;
      }
      if(sum == i)
      {
        if(flag)
        {
          System.out.println("Armstrong Numbers in the given range are ");
          flag = false;
        }
        System.out.print(i + " ");
      }
    }
    if(flag)
      System.out.println("There were no Armstrong Numbers in the given range.");
  }
}

File: C code/ArmstrongNumberRange.c
  #include<stdio.h>
  #include<math.h>

  int main(void)
  {
    int a,b,flag = 1;
    printf("Armstrong Numbers in a range\n");
    printf("^^^^^^^^^^^^^^^^^^^^^^^^\n");
    printf("Enter the range (eg: 1<space>1000) : ");
    scanf("%d %d", &a, &b);
    for(int i = a; i<= b; i++)
    {
      int sum = 0, length = 0, temp = i;
      //finding the number of digits
      for(length=0; temp != 0; length++, temp /= 10);
      temp = i;
      while(temp!=0)
      {
        sum += pow(temp%10,length);
        temp /= 10;
      }
      if(sum == i)
      {
        if(flag)
        {
          printf("Armstrong Numbers in the given range are \n");
          flag = 0;
        }
        printf("%d ", i);
      }
    }
    if(flag)
      printf("There were no Armstrong Numbers in the given range.\n");
    return 0;
  }
